rclpy/src/rclpy/_rclpy.c

```diff
@@ -2502,13 +2466,13 @@ rclpy_wait_set_add_entity(PyObject * Py_UNUSED(self), PyObject * args)
       (rclpy_subscription_t *)PyCapsule_GetPointer(pyentity, "rclpy_subscription_t");
     ret = rcl_wait_set_add_subscription(wait_set, &(sub->subscription), &index);
   } else if (0 == strcmp(entity_type, "client")) {
-    rcl_client_t * client =
-      (rcl_client_t *)PyCapsule_GetPointer(pyentity, "rcl_client_t");
-    ret = rcl_wait_set_add_client(wait_set, client, &index);
+    rclpy_client_t * client =
+      (rclpy_client_t *)PyCapsule_GetPointer(pyentity, "rclpy_client_t");
```

Comment:
All other uses of `PyCapsule_GetPointer` check for a NULL pointer after it, but this one and the one below seem to elide it.  I know that isn't new to this PR, but we should probably be more careful here.

Reply:
Check that pointers are not `NULL` before using them in 5f7b7c8